File: code/frontend/src/app/models/dto/ProdutoEdicaoDTO.ts
export interface ProdutoEdicaoDTO {
  id: number;
  nome: string;
  precoCusto: number;
  margemLucro: number;
  precoFinalVenda: number;
  idCategoria: number;
  quantidadeDisponivelEstoque: number;
  quantidadeMinimaEstoque: number;
  idUnidadeMedida: number;
  quantidadePorMedida: number;
  dataVencimento?: Date;
  codigoBarras?: string;
}

File: code/backend/src/main/java/br/com/anagropets/dto/produto/ProdutoEdicaoDTO.java
package br.com.anagropets.dto.produto;

import java.time.LocalDate;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ProdutoEdicaoDTO {
	private @NotNull Long id;
	private @NotNull @Size(max = 100) String nome;
	private @NotNull Float precoCusto;
    private @NotNull Float margemLucro;
    private @NotNull Float precoFinalVenda;
    private @NotNull Long idCategoria;
    private @NotNull Integer quantidadeDisponivelEstoque;
    private Integer quantidadeMinimaEstoque;
    private @NotNull Long idUnidadeMedida;
    private @NotNull Double quantidadePorMedida;
    private LocalDate dataVencimento;
    private String codigoBarras;
}
